examples/bluetooth/ble_remote_control/main/hidd.c

```diff
@@ -0,0 +1,548 @@
+            ESP_GATT_PERM_READ,
+            sizeof(UUID_BAT_LVL_VAL),
+            sizeof(UUID_BAT_LVL_VAL),
+            (uint8_t *) &UUID_BAT_LVL_VAL,
+        },
+    },
+    [IDX_HID_CTNL_PT_CHAR] = {
+        {ESP_GATT_AUTO_RSP},
+        {
+            ESP_UUID_LEN_16, 
+            (uint8_t *) &UUID_CHAR_DECLARE, 
+            ESP_GATT_PERM_READ,
+            sizeof(uint8_t),
+            sizeof(CHAR_PROP_READ),
+            (uint8_t *) &CHAR_PROP_READ,
+        },
+    },
+    [IDX_HID_CTNL_PT_VAL] = {
+        {ESP_GATT_AUTO_RSP},
+        {
+            ESP_UUID_LEN_16, 
+            (uint8_t *) &UUID_HID_CTNL_PT_VAL, 
+            ESP_GATT_PERM_WRITE,
+            sizeof(uint8_t),
+            sizeof(val_hid_cntl_pt),
+            (uint8_t *) &val_hid_cntl_pt,
+        },
+    },
+    [IDX_HID_REPORT_CHAR] = {
+        {ESP_GATT_AUTO_RSP},
+        {
+            ESP_UUID_LEN_16, 
+            (uint8_t *) &UUID_CHAR_DECLARE, 
+            ESP_GATT_PERM_READ,
+            sizeof(uint8_t),
+            sizeof(CHAR_PROP_READ_NOTIFY),
+            (uint8_t *) &CHAR_PROP_READ_NOTIFY,
+        },
+    },
+    [IDX_HID_REPORT_VAL] = {
+        {ESP_GATT_AUTO_RSP},
+        {
+            ESP_UUID_LEN_16, 
+            (uint8_t *) &UUID_HID_REPORT_VAL, 
+            ESP_GATT_PERM_READ,
+            sizeof(val_hid_report),
+            sizeof(val_hid_report),
+            (uint8_t *) &val_hid_report,
+        },
+    },
+    [IDX_HID_REPORT_DESCR_CCCD] = {
+        {ESP_GATT_AUTO_RSP},
+        {
+            ESP_UUID_LEN_16, 
+            (uint8_t *) &UUID_HID_CLIENT_CONFIG_DESCR, 
+            ESP_GATT_PERM_READ | ESP_GATT_PERM_WRITE,
+            sizeof(val_cccd),
+            sizeof(val_cccd),
+            (uint8_t *) &val_cccd,
+        },
+    },
+    [IDX_HID_REPORT_DESCR_RPT_REF] = {
+        {ESP_GATT_AUTO_RSP},
+        {
+            ESP_UUID_LEN_16, 
+            (uint8_t *) &UUID_HID_REPORT_REF_DESCR, 
+            ESP_GATT_PERM_READ,
+            sizeof(val_report_ref),
+            sizeof(val_report_ref),
+            (uint8_t *) &val_report_ref,
+        },
+    },
+};
+
+// HID Event Private Functions
+void hid_register_event(esp_gatt_if_t gatts_if);
+void hid_attribute_table_created_event(esp_ble_gatts_cb_param_t *param);
+void hid_close_event(void); 
+
+void hid_event_callback(esp_gatts_cb_event_t event, esp_gatt_if_t gatts_if,
+									esp_ble_gatts_cb_param_t *param)
+{    
+    switch(event) {
+    case ESP_GATTS_REG_EVT:
+        ESP_LOGI(HID_DEMO_TAG, "Profile Registered, app_id %04x, status %d",
+                 param->reg.app_id, param->reg.status);
+        hid_register_event(gatts_if);
+        break;
+    case ESP_GATTS_CONNECT_EVT:
+        ESP_LOGI(HID_DEMO_TAG, "Profile Reciving a new connection, conn_id %d", param->connect.conn_id);
+        // Save the connection ID, need this for sending notification in send_user_input()
+        client.connection_id = param->connect.conn_id; 
+        is_connected = true;
```

Comment:
Don't set `is_connected` here. Because the connection is required to be encrypted for HID profile. We should set it to true in `ESP_GAP_BLE_AUTH_CMPL_EVT`.